### ISSUE TITLE
[Firefox] Make use of the webext-specific API to clear localStorage and indexedDB

### ISSUE THREAD

**Issue by gorhill** [gorhill/uMatrix#829]
Excerpt from <https://blog.mozilla.org/addons/2017/11/20/extensions-in-firefox-58/>:

> The browsingData API now supports clearing the indexedDB storage area
> The browsingData API supports clearing localStorage by hostname, similar to cookies

Currently uMatrix uses a content script to try to clear localStorage as per cookie cell, while it does not touch indexedDB storage due to missing API to enumerate databases.

**Comment by mikhaelkh** [comment]
Are storages being cleared in Chrome? I unchecked all hosts files, purged all caches. Storage used  doesn't decrease and is ~3.3MB.

**Comment by zero77** [comment]
Yes, It would be nice to have control over allowing or blocking indexedDB storage throw the popup panel and auto clearing of indexedDB storage, if that’s what you were thinking.

Also, I think this API will be supported from the 23rd in the main FF release.
https://wiki.mozilla.org/RapidRelease/Calendar

**Comment by zero77** [comment]
This may not currently be possible in chrome but it should work well in FF.
https://bugs.chromium.org/p/chromium/issues/detail?id=78093
